deepchem/feat/molecule_featurizers/sparse_matrix_one_hot_featurizer.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from typing import List
+import numpy as np
+import scipy
+from deepchem.feat.base_classes import Featurizer
+from typing import Any, Iterable
+
+logger = logging.getLogger(__name__)
+CHARSET = [
+    'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R',
+    'S', 'T', 'V', 'W', 'Y', 'X', 'Z', 'B', 'U', 'O'
+]
+
+
+class SparseMatrixOneHotFeaturizer(Featurizer):
+  """Encodes any arbitrary string as a one-hot array.
+
+  This featurizer uses the sklearn OneHotEncoder to create
+  sparse matrix representation of a one-hot array of any string.
+  It is expected to be used in large datasets that produces memory overload
+  using standard featurizer such as OneHotFeaturizer. For example: SwissprotDataset
+
+
+  Standalone Usage:
+
+  >>> import deepchem as dc
+  >>> featurizer = dc.feat.SparseMatrixOneHotFeaturizer()
+  >>> sequence = "MMMQLA"
+  >>> encodings = featurizer.featurize([sequence])
+  >>> encodings[0].shape
+  (6, 25)
+
+  """
+
+  def __init__(self, charset: List[str] = CHARSET):
+    """Initialize featurizer.
+
+    Parameters
+    ----------
+    charset: List[str] (default code)
+      A list of strings, where each string is length 1 and unique.
+    """
+    if len(charset) != len(set(charset)):
+      raise ValueError("All values in charset must be unique.")
+    self.charset = charset
+    from sklearn.preprocessing import OneHotEncoder
+    cat = np.array(self.charset).reshape(1, len(self.charset))
+    self.ohe = OneHotEncoder(categories=list(cat), handle_unknown='ignore')
+
+  def featurize(self,
+                datapoints: Iterable[Any],
+                log_every_n: int = 1000,
+                **kwargs) -> np.ndarray:
+    """Featurize strings.
+    Parameters
+    ----------
+    datapoints: list
+      A list of either strings (str or numpy.str_)
+    log_every_n: int, optional (default 1000)
+      How many elements are featurized every time a featurization is logged.
+    """
+    if (len(datapoints) < 1):
```

Comment:
Mypy is complaining here since an iterable doesn't necessarily have a length. I think you don't need this length check here